tests/listusers/listusers_test.go

```diff
@@ -46,6 +47,7 @@ func testRunAll(t *testing.T, engine string) {
 	cfg.Log.Level = "error"
 	cfg.Datastore.Engine = engine
 	cfg.Experimentals = []string{"enable-list-users"}
+	cfg.ListUsersDeadline = 6 * time.Second
```

Comment:
> Example 2: Timeouts due to slow Github action compute resources. The solution here is to increase the timeout from 3s to 6s to allow for more reliable completion.

I don't think this is the actual root cause. The error was

> Error Trace:	/home/runner/work/openfga/openfga/tests/listusers/listusers.go:187
                            	Error:      	An error is expected but got nil.

I believe the problem is this code
https://github.com/openfga/openfga/blob/ffe0bfd2ec0c25f366e889c857240c49a11e0ee8/pkg/server/commands/listusers/list_users_rpc.go#L226-L237

If `case <-doneWithFoundUsersCh:` executes first but there was an error, you will miss it.

Reply:
Hi Maria, thanks for raising this concern.

I don't believe its possible for both the `err` and `doneWithFoundUsersCh` cases to both be satisfied simultaneously because the `foundUsersCh` gets closed after the error is checked. However, it is possible that both the `err` and context cancelled cases could be simultaneously true.

This exact test failure can be observed locally if we set the timeout arbitrarily low, which suggests that it is completely possible that this test failure is due to timeout being hit. But I agree that it doesn't appear exactly intuitive.

Speaking pragmatically, the change to extend the timeout is simple and can be easily disproven if we continue to see test failures, which _should_ be evident by now considering the frequency of failures we've seen. At which point, we can more precisely dissect the failure.

Reply:
> I don't believe its possible for both the err and doneWithFoundUsersCh cases to both be satisfied simultaneously because the foundUsersCh gets closed after the error is checked.

I did a quick test here: https://go.dev/play/p/EixeM07Zolf

In that code, the error channel gets a message before the `done` channel gets a message. And yet I don't see the error.